### PR TITLE
Let override default response size limit

### PR DESCRIPTION
**Pull request by jcanizales**
Fixes https://github.com/grpc/grpc/issues/7564

TODO: Catch the C-library error to improve the error message.

### CODE REVIEW

File: src/objective-c/GRPCClient/private/GRPCHost.m [hunk 1]
```diff
@@ -113,6 +113,12 @@ + (void)flushChannelCache {
   }
 }
 
++ (void)clearAllHostsForTesting {
```
Comment (muxi) [inline src/objective-c/GRPCClient/private/GRPCHost.m]: Suggest name it resetHostCache / clearHostCache.

Comment (jcanizales) [inline src/objective-c/GRPCClient/private/GRPCHost.m]: What about `resetHostSettings`? (It's not obvious to the user that there's a cache involved, as they just setting configs for specific hosts. I might not even be right from us to call it a cache in the implementation, as it's not really a trick to improve performance but rather a store).

Comment (muxi) [inline src/objective-c/GRPCClient/private/GRPCHost.m]: Yea that works :)

**Comment by muxi** [comment]
Note Travis is failing

**Comment by jcanizales** [comment]
Green enough?

**Comment by muxi** [comment]
LGTM

**Pull request merged**